### PR TITLE
[ops, dist, config, docs, ci] fix: make chunk loss compatible with activation offload

### PR DESCRIPTION
**Pull request by YLlllllllllll**
### What does this PR do?

> Make `chunk_loss` / `npu` cross-entropy work with activation offload.
>
> Previously, `ChunkLoss` used `torch.func.grad_and_value`, which rejects any
> active `saved_tensors_hooks` — including the hooks installed by
> `enable_activation`. Enabling both paths failed closed.
>
> This PR replaces that transform with an equivalent first-order
> `torch.autograd.grad` VJP, keeps the short-lived chunk CE graph on device,
> and lets only the long-lived precomputed grads go through the outer offload
> hooks. An opt-in `cross_entropy_loss_release_cache` flag can reclaim the
> transient CE allocator cache before model backward on memory-constrained
> profiles.

### Checklist Before Starting

- Search for relative PRs/issues and link here: none found on GitHub main; extracted from internal NPU/act-offload bring-up (kept separate from Ascend 910B4 Mojo overlay work)
- PR title follows `[{modules}] {type}: {description}` format

### Test

> - New unit tests in `tests/ops/test_chunk_loss_saved_tensor_hooks.py`:
>   - chunk loss under synthetic saved-tensor hooks preserves loss/grads
>   - production `chunk_loss_function` under `custom_save_on_cpu`
>   - opt-in `release_cache` runs only after outer pack completes
>   - factory binds `cross_entropy_loss_release_cache`
> - `python3 -m py_compile` on touched Python files
> - Validated in internal NPU training profiles that combine
>   `chunk_loss`/`npu` with `train.accelerator.offload_config.enable_activation`

### API and Usage Example

```yaml
model:
  ops_implementation:
    cross_entropy_loss_implementation: chunk_loss   # or npu
    cross_entropy_loss_release_cache: false          # set true on tight HBM profiles
train:
  accelerator:
    offload_config:
      enable_activation: true
```

### Design & Code Changes

1. **`veomni/ops/kernels/cross_entropy/chunk_loss.py`**
   - Drop `torch.func`; compute per-chunk VJP with `torch.autograd.grad`
   - Shadow outer activation-offload hooks with identity hooks for the
     short-lived CE graph so vocab-sized logits are not forced through CPU
   - After `ChunkLoss.apply`, optionally sync + `empty_cache` when
     `release_cache=true`
2. **`veomni/ops/kernels/cross_entropy/__init__.py`**
   - Thread `cross_entropy_loss_release_cache` from ops config into the
     chunk-loss dispatch
3. **`veomni/arguments/arguments_types.py` + docs**
   - Document the new opt-in flag
4. **Tests**
   - Regression coverage for hooks / offload / release_cache

This change is hardware-agnostic and does **not** include Ascend 910B4 Mojo
compat overlays, validators, or GDR pad logic.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks (targeted compile + diff --check)
- Added/updated documentation
- Added tests (`tests/ops/test_chunk_loss_saved_tensor_hooks.py`)

### CODE REVIEW

**Comment by YLlllllllllll** [comment]
@Luosuu All CI checks passed on `eba3837` (3 green) — ready to merge when you have a moment.

**Review by Coach257 — APPROVED** [review]

**Comment by YLlllllllllll** [comment]
@Coach257 CI is all green and your approval is already on `eba3837` — ready to merge when you have a moment. Thanks!

**Comment by YLlllllllllll** [comment]
Closed accidentally while recreating the head branch to rebase past the fork's ByteDance-author push hook. Rebased tip is now in **#1004** (same change, on latest `main`).
